src/main/java/fi/metatavu/airquality/GenericFileConfigurationParser.java

```diff
@@ -0,0 +1,87 @@
+package fi.metatavu.airquality;
+
+import com.google.gson.Gson;
+import fi.metatavu.airquality.configuration_parsing.GenericFileConfiguration;
+import fi.metatavu.airquality.configuration_parsing.ParameterType;
+import fi.metatavu.airquality.configuration_parsing.RequestParameters;
+import org.opentripplanner.datastore.DataSource;
+
+import java.io.InputStreamReader;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+public class GenericFileConfigurationParser {
+
+	/**
+	 * Parse settings.json data source, verify the contents and return the parsed GenericFileConfiguration[]
+	 * @param dataSource dataSource from settings.json file
+	 * @return GenericFileConfiguration[]
+	 */
+	public static GenericFileConfiguration[] parse(DataSource dataSource) {
+		if (dataSource.exists()) {
+			GenericFileConfiguration[] configurations = new Gson().fromJson(new InputStreamReader(dataSource.asInputStream()),
+							GenericFileConfiguration[].class);
+			//validate the configurations
+			for (GenericFileConfiguration configuration : configurations){
+				RequestParameters[] requestParametersList = configuration.getRequestParameters();
+				if (!areRequestParamValid(Arrays.asList(requestParametersList)))
+					throw new IllegalArgumentException("The settings file has incorrect request parameters");
+			}
+			return configurations;
+		}
+		return null;
+	}
+
+	/**
+	 * todo check if this is still needed?
+	 * @param requestParametersList
+	 * @return
+	 */
+	public static boolean areRequestParamValid(List<RequestParameters> requestParametersList) {
+		AtomicBoolean res = new AtomicBoolean(true);
+		requestParametersList.stream()
+						.collect(Collectors.groupingBy(RequestParameters::getVariable))
+						.forEach((k, v) -> {
+							if (v.size() != 2)
+								res.set(false);
+						});
+		return res.get();
```

Comment:
Follow todo in the Java docs above. If this is needed - update Java docs with correct description. If not needed - delete.